Terminal.Gui/ConsoleDrivers/NetDriver/NetEvents.cs

```diff
@@ -33,41 +32,15 @@ public NetEvents (ConsoleDriver consoleDriver)
 
     public InputResult? DequeueInput ()
     {
-        while (_inputReadyCancellationTokenSource != null
-               && !_inputReadyCancellationTokenSource.Token.IsCancellationRequested)
+        while (_inputReadyCancellationTokenSource != null)
         {
             _waitForStart.Set ();
             _winChange.Set ();
 
-            try
-            {
-                if (!_inputReadyCancellationTokenSource.Token.IsCancellationRequested)
-                {
-                    if (_inputQueue.Count == 0)
-                    {
-                        _inputReady.Wait (_inputReadyCancellationTokenSource.Token);
-                    }
-                }
-            }
-            catch (OperationCanceledException)
-            {
-                return null;
-            }
-            finally
-            {
-                _inputReady.Reset ();
-            }
-
 #if PROCESS_REQUEST
             _neededProcessRequest = false;
 #endif
-            if (_inputQueue.Count > 0)
-            {
-                if (_inputQueue.TryDequeue (out InputResult? result))
-                {
-                    return result;
-                }
-            }
+            return _inputQueue.Take(_inputReadyCancellationTokenSource.Token);
```

Comment:
Because its a blocking collection this will block when there are 0 items until items get added then it will resolve.